### PR TITLE
[#3558] Docs for Contributing and Defining Components 

### PR DESCRIPTION
**Pull request by armanjindal**
The purpose of this PR is to organize discussions around how we want to maintain and update components. I have tried to capture our internal discussion on this issue in this initial draft. This PR also includes an update to our definition of `Components` in the glossary (#3294). Please amend/discuss/comment!

The secondary purpose is to merge these docs **before** we implement the proposed change. This way, we can adopt a "docs-driven" development approach.

The primary change proposed is the creation of an external repository where the Helm package and information on components are stored. The benefits of this approach are: 

1. We can update information on components between release cycles (if needed). 
2. The `description.yaml` associated with a connector file becomes the "source of truth" on components and can be standardized for the Control Center.
3. The helm files can help potential enterprise clients technically audit us.
4. We begin to create the infrastructure to support third-party components.

### CODE REVIEW

File: docs/docs/guides/contributing-components.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+sidebar_label: Contributing Components
+---
+
+:::warning
+
+This functionality is under development. With these docs, we aim to elicit feedback from our community (you!) and focus our development efforts. Currently, we do not offer support for 3rd party components, but we are working toward supporting this.
+
+:::
+
+`Airy Core` is fundamentally a collection of components. When you [install `Airy Core`](../getting-started/installation/introduction.md), you install a set of _core_ components which create the minimum infrastructure to install other components and begin using the platform.
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:12]: `Airy Core` is not really how we market the open source. Let's just call it `Airy` and mention that it shipds with a few components out-of-the-box.

File: docs/docs/guides/contributing-components.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+6. frontend-control-center
+
+Airy's real value comes from the additional components you install through our marketplace in the Control Center UI of your Airy Instance. This documentation is on the processes we follow to create, update and store our components.
+
+## Components
+
+Airy is designed to support a plug-and-play architecture. Components are installed and run as pods in your cluster. Components can provide four kinds of functionality:
+
+1. Connector - i.e Rasa, DialogFlow
+2. Source - i.e., Facebook, WhatsApp, SMS
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:30]: We decided to not differentiate between `sources` and `connectors` on a component level. 

In the [glossary docs](https://docs.google.com/document/d/1MjLF1eylIgMIhxzP7kJHHroRQXeKQyO4-824MR0pplE/edit) I suggested the following definition for a connector which nicely brackets both use cases:
>  A connector is a type of component that authenticates with one or more third-party services in order to exchange data

This document should link to the glossary definitions that aren't here yet, unfortunately. @AitorAlgorta perhaps we can already add definitions for `component` and `connector` in this PR?

File: docs/docs/guides/contributing-components.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+3. API - i.e
+4. User Interface - i.e
+
+As a technical artifact, each component is a containerized application that is packaged and can be installed. We use Docker and Helm for this.
+
+The Helm package and information on each components (except core components) are stored in an external repository managed by Airy called [airy-components](https://github.com/airyhq/airy-/airy-components). This repository is made up directories where each one contains the component's Helm package and information on it.
+
+## The Component File Structure
+
+Below is a model of the file stucture of a single component inside the [`airy-components`](https://github.com/airyhq/airy-/airy-components) repository.
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:40]: I created the repository that we can use to link to correctly: https://github.com/airyhq/airy-components

File: docs/docs/guides/contributing-components.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+```
+/airy-components
+	/[COMPONENT_NAME]
+		information.yaml
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:45]: `description.yaml` seems a bit more apt. What do you think? 🤔 

Also sharing this neat little app for generating these file graphs: https://tree.nathanfriend.io/

File: docs/docs/guides/contributing-components.md (hunk 3 above)
Comment (armanjindal) [reply docs/docs/guides/contributing-components.md:40]: Amazing!

File: docs/docs/guides/contributing-components.md (hunk 4 above)
Comment (armanjindal) [reply docs/docs/guides/contributing-components.md:45]: Agreed.

File: docs/docs/guides/contributing-components.md (hunk 1 above)
Comment (armanjindal) [reply docs/docs/guides/contributing-components.md:12]: Sounds good. Although there are a ton of references to Airy Core in the docs, so we might need a larger fix. For now I will refer to it as Airy.

File: docs/docs/guides/contributing-components.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+These core components are:
+
+1. airy-controller
+2. api-admin
+3. api-communication
+4. api-websocket
+5. frontend-inbox
+6. frontend-control-center
+
+Airy's real value comes from the additional components you install through our marketplace in the Control Center UI of your Airy Instance. This documentation is on the processes we follow to create, update and store our components.
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:23]:
```suggestion
Airy's real value comes from the additional components you install through our catalog in the Control Center UI of your Airy Instance. 
In the following, we will explain how to create, update and store components.
```

File: docs/docs/guides/contributing-components.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+import ComponentDefinition from '../getting-started/componentDefinition.mdx'
+
+<ComponentDefinition/>
+
+The Helm package and information on each components (except core components) are stored in an external repository managed by Airy called [airy-components](https://github.com/airyhq/airy-/airy-components). This repository is made up directories where each one contains the component's Helm package and information on it.
+
+## The Component File Structure
+
+Below is a model of the file stucture of a single component inside the [`airy-components`](https://github.com/airyhq/airy-/airy-components) repository.
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:35]:
```suggestion
Below is a model of the file structure of a single component inside the [`airy-components`](https://github.com/airyhq/airy-components) repository.
```

File: docs/docs/guides/contributing-components.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+Airy's real value comes from the additional components you install through our marketplace in the Control Center UI of your Airy Instance. This documentation is on the processes we follow to create, update and store our components.
+
+## Components
+
+import ComponentDefinition from '../getting-started/componentDefinition.mdx'
+
+<ComponentDefinition/>
+
+The Helm package and information on each components (except core components) are stored in an external repository managed by Airy called [airy-components](https://github.com/airyhq/airy-/airy-components). This repository is made up directories where each one contains the component's Helm package and information on it.
```
Comment (chrismatix) [inline docs/docs/guides/contributing-components.md:31]:
```suggestion
The Helm package and information on each component (except core components) are stored in an external repository managed by Airy called [airy-components](https://github.com/airyhq/airy-/airy-components). This repository is made up of directories where each one contains the component's Helm package and information on it.
```

File: docs/docs/getting-started/componentDefinition.mdx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+A component is a Helm Chart that can be installed on an Airy Cluster in order to extend its functionality.
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/componentDefinition.mdx:1]: We do call components also the components which are part of `Airy Core`: `api-admin`, `api-communication` :thinking: Perhaps a collection of functionalities and resources? There some components which are part of Core and the others can be installed on top.

File: docs/docs/getting-started/componentDefinition.mdx [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+Components can be characterized by the functionality they enable:
+
+1. UI Components that provide new user interfaces in Airy
+2. Connector Components that connect one or more third-party services with Airy to exchange data. 
+3. API Components that expose additional endpoints to Airy.
+
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/componentDefinition.mdx:8]: The are also `integration` components. Such as the `integration-webhook`.

File: docs/docs/getting-started/componentDefinition.mdx [hunk 10]
```diff
@@ -0,0 +1,13 @@
+A component is a Helm Chart that can be installed on an Airy Cluster in order to extend its functionality.
+
+Components can be characterized by the functionality they enable:
+
+1. UI Components that provide new user interfaces in Airy
+2. Connector Components that connect one or more third-party services with Airy to exchange data.
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/componentDefinition.mdx:6]: As `sources` is everywhere in the docs, I recommend that we add a note here that the `sources` fit also under connectors.

File: docs/docs/getting-started/componentDefinition.mdx [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+Components can be characterized by the functionality they enable:
+
+1. UI Components that provide new user interfaces in Airy
+2. Connector Components that connect one or more third-party services with Airy to exchange data. 
+3. API Components that expose additional endpoints to Airy.
+
+:::note
+
+A Component can provide multiple functions (e.g., Connector & UI or UI & API) but usually has a single primary function. You can find a more detailed description of a specific component in the `description.yaml` file of the component in our [airy-components](https://github.com/airyhq/airy-components) respository.
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/componentDefinition.mdx:11]: I agree that this is how we should do it (with the `description.yaml` file). However this is not there at the moment and the CI pipeline should be setup in place before we can write this.

File: docs/docs/guides/contributing-components.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+The Helm package and information on each component (except core components) are stored in an external repository managed by Airy called [airy-components](https://github.com/airyhq/airy-/airy-components). This repository is made up of directories where each directory contains a component's Helm package and information.
+
+## The Component File Structure
+
+Below is a model of the file structure of a single component inside the [`airy-components`](https://github.com/airyhq/airy-components) repository.
+
+```
+airy-components/
+└── [COMPONENT_NAME]/
+    ├── information.yaml
```
Comment (ljupcovangelski) [inline docs/docs/guides/contributing-components.md:39]: I think it was `description.yaml` in the previous text?

Comment (armanjindal) [reply docs/docs/guides/contributing-components.md:39]: Yes! Good catch. Will fix

File: docs/docs/getting-started/componentDefinition.mdx (hunk 11 above)
Comment (armanjindal) [reply docs/docs/getting-started/componentDefinition.mdx:11]: We will hold off on this detail in the definition and move it to the contributing components page. This is marked as under development, so will help to implement when we do this.

File: docs/docs/getting-started/componentDefinition.mdx (hunk 10 above)
Comment (armanjindal) [reply docs/docs/getting-started/componentDefinition.mdx:6]: We will omit these details about the **kinds** of connectors for now. As we discussed, this is something for a more extensive discussion between product and engineering to unify our language.

File: docs/docs/getting-started/componentDefinition.mdx (hunk 9 above)
Comment (armanjindal) [reply docs/docs/getting-started/componentDefinition.mdx:8]: Same as a comment below. We will defer.

**Review by ljupcovangelski — DISMISSED** [review]
Great initiative and a good start of the "contribution of components" docs :+1:

**Review by ljupcovangelski — APPROVED** [review]

**Review by ljupcovangelski — APPROVED** [review]

**Pull request merged**